packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -59,16 +60,27 @@ async function createDataObject<TObj extends PureDataObject, I extends DataObjec
 
     // request mixin in
     runtimeClass = mixinRequestHandler(
-        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) =>
-            (await PureDataObject.getDataObject(runtimeArg)).request(request),
-            runtimeClass);
+        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) => {
+            const router = (await runtimeArg.IFluidHandle?.get() as FluidObject<IFluidRouter>).IFluidRouter;
+            assert(router !== undefined, "Data store runtime handle is not an IFluidRouter");
+            return router.request(request);
+        },
+        runtimeClass);
 
     // Create a new runtime for our data store
     // The runtime is what Fluid uses to create DDS' and route to your data store
-    const runtime = new runtimeClass(
+    const runtime: FluidDataStoreRuntime = new runtimeClass(
         context,
         sharedObjectRegistry,
         existing,
+        async (rt: IFluidDataStoreRuntime) => {
+            assert(instance !== undefined, "Entrypoint is undefined");
+            // Calling finishInitialization here like PureDataObject.getDataObject does to keep the same behavior,
+            // since accessing the runtime's entrypoint is how we want the data object to be retrieved going forward.
+            // Without this I ran into issues with the load-existing flow not working correctly.
+            await instance.finishInitialization(true);
```

Comment:
In the current version of the code, `PureDataObject.getDataObject()` is executed for every request, and it in turn calls `finishInitialization(true)` on the data object, so that call could happen several times. In the new version this should only get called once, the first time the entrypoint is initialized. That could happen during the first request, when the new code gets the runtime's entrypoint through its `IFluidHandle`, which will trigger the LazyPromise that runs this lambda.

Overall it feels to me like that's a win, but wondering if there are weird side-effects if we now only call this once.